script/tool/lib/src/common/plugin_command.dart

```diff
@@ -76,10 +76,11 @@ abstract class PluginCommand extends Command<void> {
   /// This can be overridden for testing.
   final ProcessRunner processRunner;
 
-  /// The git directory to use. By default it uses the parent directory.
+  /// The git directory to use. If unset, [getGitDir] uses the packages
+  /// directory's enclosing repository.
   ///
   /// This can be mocked for testing.
-  final GitDir? gitDir;
+  GitDir? _gitDir;
```

Comment:
This should be a late final non-null field, no?

Reply:
It's used as a cache, so cannot be late final.